### PR TITLE
fix: fix python requirements.

### PR DESCRIPTION
**Pull request by attiyaIshaque**
**Description:** The python requirements has been failing for some time now.
**JIRA:** https://2u-internal.atlassian.net/browse/VAN-1161

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: pylintrc [hunk 1]
```diff
@@ -261,7 +261,6 @@ disable =
 	no-else-return,
 	invalid-name,
 	fixme,
-	no-self-use,
```
Comment (zainab-amir) [inline pylintrc:264]: why are we removing this?

File: edx_ace/channel/braze.py [hunk 2]
```diff
@@ -135,7 +135,7 @@ def deliver(self, message, rendered_message):
 
         # https://www.braze.com/docs/api/endpoints/messaging/send_messages/post_send_messages/
         # https://www.braze.com/docs/api/objects_filters/email_object/
-        response = requests.post(
+        response = requests.post(  # pylint: disable=missing-timeout
```
Comment (zainab-amir) [inline edx_ace/channel/braze.py]: Is there a reason we have not added timeout?

File: pylintrc (hunk 1 above)
Comment (attiyaIshaque) [reply pylintrc:264]: In new version of pylint it has been removed. https://pylint.pycqa.org/en/latest/whatsnew/2/2.14/summary.html#removed-checkers.

File: edx_ace/channel/braze.py (hunk 2 above)
Comment (attiyaIshaque) [reply edx_ace/channel/braze.py]: We can add it but how much time we can add like 10 or 5?

File: pylintrc (hunk 1 above)
Comment (zainab-amir) [reply pylintrc:264]: It has been made optional but not completely removed

File: edx_ace/channel/braze.py (hunk 2 above)
Comment (zainab-amir) [reply edx_ace/channel/braze.py:138]: Lets discuss this on Monday

File: edx_ace/tests/test_message.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 LOG = logging.getLogger(__name__)
 
 context_values = st.one_of(st.text(), st.booleans(), st.floats(allow_nan=False))
-dates = st.datetimes(timezones=st.none() | timezones() | st.none())
+dates = st.datetimes(timezones=st.none() or timezones() or st.none())
```
Comment (syedsajjadkazmii) [inline edx_ace/tests/test_message.py]: The operator type is changed completely from a bitwise OR to a relational OR operator, which would change the logic. Could you please tell me what the purpose of this line was and how the operator change does not affect the purpose of this line?

File: edx_ace/channel/braze.py [hunk 4]
```diff
@@ -20,6 +20,7 @@
 LOG = logging.getLogger(__name__)
 
 NEXT_ATTEMPT_DELAY_SECONDS = 30
+BRAZE_TIMEOUT = 10
```
Comment (zainab-amir) [inline edx_ace/channel/braze.py]:
```suggestion
BRAZE_API_TIMEOUT = 5
```

File: edx_ace/channel/braze.py [hunk 5]
```diff
@@ -156,6 +157,7 @@ def deliver(self, message, rendered_message):
                     },
                 },
             },
+            timeout=BRAZE_TIMEOUT
```
Comment (zainab-amir) [inline edx_ace/channel/braze.py]: If anyone wants to override the value, they should be able to use the django settings to do so.
```suggestion
            timeout=getattr(settings, 'ACE_DEFAULT_API_TIMEOUT', BRAZE_API_TIMEOUT)
```

**Review by zainab-amir — COMMENTED** [review]
Waiting for your reply on the removal of `no-self-use` from pylint config.

**Comment by attiyaIshaque** [comment]
> Waiting for your reply on the removal of `no-self-use` from pylint config.

IMO We should remove it because there is no quality failure regarding this.

**Comment by zainab-amir** [comment]
Apologies I missed that it was in the disabled settings in pylintrc. We can remove it

File: edx_ace/tests/test_message.py (hunk 3 above)
Comment (attiyaIshaque) [reply edx_ace/tests/test_message.py]: @syedsajjadkazmii Thanks for pointing out, yup it changed the logic. ` st.datetimes(timezones=st.none() or timezones() or st.none())` it only returns one value none. 
so i have updated the code which does not affect the logic.

**Review by syedsajjadkazmii — APPROVED** [review]

File: edx_ace/tests/test_message.py (hunk 3 above)
Comment (syedsajjadkazmii) [reply edx_ace/tests/test_message.py]: @attiyaIshaque thanks for clerification

**Pull request merged**